crypto/helpers.go

```diff
@@ -104,45 +132,98 @@ func AESDecrypt(ciphertext []byte, iv []byte, key []byte) ([]byte, error) {
 	return UnPad(paddedPlaintext), nil
 }
 
-func GenerateRSAKey() *rsa.PrivateKey {
+func GenerateRSAKey() (*rsa.PrivateKey, error) {
 	if key, err := rsa.GenerateKey(rand.Reader, 2048); err != nil {
-		panic(fmt.Sprintf("Can't create RSA keys: %s", err.Error()))
+		return nil, fmt.Errorf("Can't create RSA keys: %s", err)
 	} else {
-		return key
+		return key, nil
 	}
 }
 
-func PemEncodeRSAPrivate(key *rsa.PrivateKey) []byte {
-	der := x509.MarshalPKCS1PrivateKey(key)
-	b := &pem.Block{Type: "RSA PRIVATE KEY", Bytes: der}
-	return pem.EncodeToMemory(b)
+func GenerateECKey() (*ecdsa.PrivateKey, error) {
+	if key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader); err != nil {
```

Comment:
It will require an additional dependency, but using ECC already adds some new dependencies.